more_itertools/more.py

```diff
@@ -111,6 +112,78 @@ def chunked(iterable, n):
     return iter(partial(take, n, iter(iterable)), [])
 
 
+def iterchunked(iterable, n):
+    # type: (typing.Iterable, int) -> typing.Iterable[typing.Iterable]
+    """Break *iterable* into iterators of (at most) length *n*:
+
+        >>> list(list(chunk) for chunk in iterchunked([1, 2, 3, 4, 5, 6], 3))
+        [[1, 2, 3], [4, 5, 6]]
+
+    If the length of *iterable* is not evenly divisible by *n*, the last
+    returned iterable will be shorter:
+
+        >>> list(list(chunk) for chunk in iterchunked([1, 2, 3, 4, 5, 6, 7, 8], 3))
+        [[1, 2, 3], [4, 5, 6], [7, 8]]
+
+    To use a fill-in value instead, see the :func:`grouper` recipe.
+
+    To iterate over lists instead of sub-iterators, see the :func:`chunked` recipe.
+
+    """
+
+    # Make sure we are working with an iterator
+    if not (
+        hasattr(iterable, '__next__') and
+        callable(iterable.__next__)
+    ):
+        iterable = iter(iterable)
+
+    # Define functions for yielding sub-iterators
+
+    def yield_and_dequeue(queue, limit):
+        # type: (Iterable, int) -> Iterable
+        """
+        De-queues and yields at most `limit` number of items from an iterator
+        """
+        for i in range(limit):
+            try:
+                yield next(queue)
+            except StopIteration:
+                break
+
+    def dequeue_and_yield(queue, limit):
+        # type: (Iterable, int) -> Iterable
+        """
+        This function first checks the next value in the queue to make sure we haven't already reached the end of the
+        iterator, *then* de-queues and yields at most `limit` number of items from the iterator--but only if there is at
+        least one upcoming value. This is to prevent the parent function from yielding empty iterators.
+        """
+        if not limit:
```

Comment:
Might it be better to check whether the argument is > 0 at the top of the function and omit this branch?

Reply:
Scratch that; I see the `limit - 1` below.